scripts/core/editor3/components/Editor3.jsx

```diff
@@ -39,39 +41,150 @@ export class Editor3 extends React.Component {
+    }
 
-    /** Handle the editor get focus event */
-    focus() {
-        this.refs.editor.focus();
+    componentWillUnmount() {
+        window.removeEventListener('contextmenu', this.onContextMenu);
     }
 
     /** Handle the editor state has been changed event*/
     onChange(editorState) {
-        this.setState({editorState});
+        this.setState({editorState}, _.debounce(this.handleSpelchecker, 100));
         this.parentOnChange(stateToHTML(editorState.getCurrentContent()));
     }
 
+    /** Handle the editor get focus event */
+    focus() {
+        this.refs.editor.focus();
+    }
+
+    /** Handle the editor tab key pressed event */
+    onTab(e) {
+        const maxDepth = 4;
+
+        this.onChange(RichUtils.onTab(e, this.state.editorState, maxDepth));
+    }
+
+    /** Handle the editor context menu */
+    onContextMenu(e) {
+        const {editorState} = this.state;
+        const contentState = editorState.getCurrentContent();
+        const selection = editorState.getSelection();
+        const block = contentState.getBlockForKey(selection.getStartKey());
+        const text = block.getText();
+
+        if (!this.isActiveSpellchecker() || !selection.getHasFocus() || !text) {
+            return;
+        }
+
+        const editorRect = ReactDOM.findDOMNode(this.refs.editor).getBoundingClientRect();
+
+        this.spellchecker.showContextMenu(text, selection,
+            e.clientX - editorRect.left - 20, e.clientY - editorRect.top + 20);
+        e.preventDefault();
+    }
+
+    /** If spellchecker is active, closes the right menu */
+    closeContextMenu() {
+        if (!this.isActiveSpellchecker()) {
+            return;
+        }
+        this.spellchecker.closeContextMenu();
+    }
+
+    /** Called by a context menu after it is opened */
+    onOpenContextMenu() {
+        this.setState({showContextMenu: true});
+    }
+
+    /** Called by a context menu after it is closed */
+    onCloseContextMenu() {
+        this.focus();
+        this.setState({showContextMenu: false});
+    }
+
+    /** Return the rendering of the context menu */
+    getContextMenu() {
+        if (!this.isActiveSpellchecker()) {
+            return '';
+        }
+        return this.spellchecker.getContextMenu();
+    }
+
+    /** Replace the word at selection with the new word */
+    replaceWord(word, textSelection) {
+        const {editorState} = this.state;
+        var newContentState;
+        var newEditorState;
+
+        this.focus();
+
+        newContentState = Modifier.replaceText(editorState.getCurrentContent(), textSelection, word);
+        newEditorState = EditorState.push(editorState, newContentState, 'spellchecker replace word');
+
+        this.onChange(newEditorState);
+    }
+
+    /** Return if the spellchecker is initialized/active */
+    isActiveSpellchecker() {
+        return !!this.spellchecker;
+    }
+
+    /** Spelcheck the text for the current block */
+    handleSpelchecker() {
```

Comment:
There is a slight spelling mistake here int he name of the method:
`s/handleSpelchecker/handleSpellcheker`